File: java-library/src/main/java/com/gblninja/tag/type/GblHeader.java
package com.gblninja.tag.type;


import com.gblninja.tag.GblType;
import com.gblninja.tag.Tag;
import com.gblninja.tag.TagHeader;
import com.gblninja.tag.TagWithHeader;

public class GblHeader implements Tag, TagWithHeader {
    private final TagHeader tagHeader;
    private final GblType tagType;
    private final long version;
    private final long gblType;
    private final byte[] tagData;

    public GblHeader(TagHeader tagHeader, GblType tagType, long version, long gblType, byte[] tagData) {
        this.tagHeader = tagHeader;
        this.tagType = tagType;
        this.version = version;
        this.gblType = gblType;
        this.tagData = tagData;
    }

    @Override
    public GblType getTagType() {
        return tagType;
    }

    @Override
    public Tag copy() {
        return new GblHeader(tagHeader, tagType, version, gblType, new byte[0]);
    }

    @Override
    public TagHeader getTagHeader() {
        return tagHeader;
    }

    @Override
    public byte[] getTagData() {
        return tagData;
    }

    public long getVersion() {
        return version;
    }

    public long getGblType() {
        return gblType;
    }
}

File: gbl/src/main/java/com/onix/gbl/parser/type/GblHeader.kt
package parser.data.type

import parser.data.tag.TagHeader
import parser.data.tag.TagInterface

data class GblHeader(
    override val tagHeader: TagHeader,
    override val tagType: GblType = GblType.HEADER_V3,
    val version: UInt,
    val gblType: UInt,
    override val tagData: ByteArray
): TagInterface {
    override fun copy(): TagInterface {
        return GblHeader(
            tagHeader = tagHeader,
            version = version,
            gblType = gblType,
            tagData = arrayOf<Byte>().toByteArray()
        )
    }
}
